Modules/Filtering/LabelMap/include/itkStatisticsLabelMapFilter.hxx

```diff
@@ -177,15 +177,27 @@ StatisticsLabelMapFilter<TImage, TFeatureImage>::ThreadedProcessLabelObject(Labe
   }
 
   // the median
-  double median = 0;
-  double count = 0; // will not be fully set, so do not use later !
+  double median = 0.0;
+  double count = 0.0; // will not be fully set, so do not use later !
   for (SizeValueType i = 0; i < histogram->Size(); ++i)
   {
     count += histogram->GetFrequency(i);
 
     if (count >= (totalFreq / 2))
     {
       median = histogram->GetMeasurementVector(i)[0];
+      // If there are an even number of elements average with the next bin with elements
+      if (labelObject->Size() % 2 == 0 && count == totalFreq / 2)
+      {
+        while (++i < histogram->Size())
+        {
+          if (histogram->GetFrequency(i) > 0)
+          {
+            median += histogram->GetMeasurementVector(i)[0];
+            median *= 0.5;
```

Comment:
shouldn't there be `break;` here?

Reply:
good catch! Thanks!

Reply:
I changed the test case to have 4 elements to cover this case.